gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2

```diff
@@ -368,15 +368,18 @@ class {{ service.async_client_name }}:
         {% if method.field_headers %}
         # Certain fields should be provided within the metadata header;
         # add these here.
-        metadata = tuple(metadata) + (
-            gapic_v1.routing_header.to_grpc_metadata((
-                {% for field_header in method.field_headers %}
-                {% if not method.client_streaming %}
-                ("{{ field_header.raw }}", request.{{ field_header.disambiguated }}),
-                {% endif %}
-                {% endfor %}
-            )),
-        )
+        metadata = tuple(metadata)
+        if all(m[0] != gapic_v1.routing_header.ROUTING_METADATA_KEY for m in metadata):
```

Comment:
Please can you add comments to clarify the reason for the additional logic?

Reply:
I updated the comment above this block to make it more clear, let me know what you think